bin/clover/src/pipelines/azure/spec.ts

```diff
@@ -600,17 +625,107 @@ function extractPropertiesFromResponseBody(
   }
   if (!("id" in result.properties)) {
     throw new Error(
-      `No id property in GET response: ${
-        operation.operationId
-      }\n\n${util.inspect(operation, { depth: 12 })}\n\n${util.inspect(
-        azureProp,
-        { depth: 12 },
-      )}`,
+      `No id property in GET response: ${operation.operationId}\n\n${
+        util.inspect(operation, { depth: 12 })
+      }\n\n${
+        util.inspect(
+          azureProp,
+          { depth: 12 },
+        )
+      }`,
     );
   }
   return result;
 }
 
+/// Applies discriminator handling by finding subtypes and adding them as new sub-objects.
+/// Recursively processes the schema tree to find and expand all discriminated types.
+function applyDiscriminators(
```

Comment:
I bet most of this method would go away if we did this inside the normalizer--populate the definition map before doing any normalizing, then let the existing recursion do its work and check for `discriminator` as we normalize a prop.

Not a necessity even if it's 100% true though--I think this will work, and we don't necessarily need access to the more advanced stuff the normalizer does (like unions) because you're nesting each object into its own sub-object, so merging (union/intersect) isn't an issue.

Reply:
You are probably right, but I was wary of further complicating the logic there. If it's the right call to do that, we can.